packages/auth/lib/CSRFProtection.ts

```diff
@@ -0,0 +1,73 @@
+import { sha256 } from "./crypto";
+import { URL } from "url";
+import { ILogger } from "@fullstack-one/logger";
+
+export class CSRFProtection {
+  private logger: ILogger;
+  private authConfig;
+
+  constructor(logger: ILogger, authConfig) {
+    this.logger = logger;
+    this.authConfig = authConfig;
+  }
+
+  public async createSecurityContext(ctx, next) {
+    ctx.securityContext = {
+      isBrowser: true,
+      isApiClient: false,
+      clientIdentifier: null,
+      requestMethod: ctx.request.method.toUpperCase(),
+      sameOriginApproved: {
+        byReferrer: false,
+        byOrigin: false
+      }
+    };
+
+    // Generate clientIdentifier for refresh-token
```

Comment:
We could remove the comment if we had a function: `private generateClientIdentifierForRefreshToken(ctx): void`

Reply:
I agree with that. However, you have to jump through code to read it because of three simple lines.

Furthermore, we would have more LoC with that function.

Reply:
Ok.